### ISSUE TITLE
Windows minion custom grains 'dict object' has no attribute

### ISSUE THREAD

**Issue by belawaeckerlig** [saltstack/salt#14237]
Hi,

I have a custom grain module in the _grains folder in the files_root of the master. After executing a highstate on a windows minion, I see, that the grain module is in the cache directory. The highstate runs fine, and not uses the grain module. But when calling a single state, which uses the grain, i get the following error:

```
    Data failed to compile:
----------
    Rendering SLS "base:win.git.checkout" failed: Jinja variable 'dict object' has no attribute 'prodenv_code_is_write_protected'; line 1

---
{% if not grains['prodenv_code_is_write_protected'] %}    <======================
get-certificate:
  file.managed:
    - name: 'c:\Users\Administrator\.ssh\github_teamcity_rsa'
    - source: salt://secrets/ssh/github_teamcity_rsa
    - makedirs: True
[...]
---
```

even after executing a sync_all, still the same error occures.

grain module:
```
def prodenv_code_is_write_protected():
    """if code is read only or mapped through vagrant into system return True"""
    isVagrantMapped = False
    testfilePath = os.path.join(_prodenv_code_path(), "writetestfile.txt")
    try:
        with open(testfilePath, "w") as f:
            f.write("writetest")
        os.unlink(testfilePath)
        isVagrantMapped = _prodenv_runs_in_vagrant()
    except IOError as e:
        if e.errno == 30:
            isVagrantMapped = True
    return {"prodenv_code_is_write_protected": isVagrantMapped}
```

state file:
```
{% if not grains['prodenv_code_is_write_protected'] %}
get-certificate:
  file.managed:
    - name: 'c:\Users\Administrator\.ssh\github_teamcity_rsa'
    - source: salt://secrets/ssh/github_teamcity_rsa
    - makedirs: True

get-ssh-config:
  file.managed:
    - name: 'c:\Users\Administrator\.ssh\config'
    - source: salt://win/ssh/config
    - makedirs: True
{% endif %}
```

Windows Server 2008 R2 Minion:
```
           Salt: 2014.1.7
         Python: 2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]
         Jinja2: 2.7.1
       M2Crypto: 0.21.1
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6
         PyYAML: 3.11
          PyZMQ: 14.1.1
            ZMQ: 4.0.4
```

Ubuntu Server 14.04 LTS Master:
```
           Salt: 2014.1.7
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
            ZMQ: 4.0.4
```

**Comment by obestwalter** [comment]
@UtahDave @basepi I would reckon this can be closed as resolved since 2015.5.3-2 as we use a proper Python installation now.

**Comment by basepi** [comment]
Thanks!
